### PR TITLE
[Navigation-Next] Add nav group enabled in chrome service

### PR DESCRIPTION
**Pull request by wanglam**
### Description

The #7060  already introduce a new use case interface in chrome service. In this PR, we will add a new advance settings `navGroupEnabled` and expose a `getNavGroupEnabled ` in chrome service.
This `useCaseEnabled` flag will be used to control whether group navigation links using use cases. If the value is `true`, all the navigation links will be grouped by use case. 
This PR also expose a `getNavGroupEnabled` method to control chrome service `setup` and `start` method's return value. Other plugin can use this method to add some specific logic for use case. 

### Issues Resolved

#7073 

## Screenshot

image

### For advanced settings changes:
- Clone branch and run `yarn osd bootstrap`
- Run `yarn start --no-base-path` and login
- Go to [advanced settings page](http://localhost:5601/app/management/opensearch-dashboards/settings)
- Toggle useCaseEnabled button to switch different values like below image:
image

### For chrome service changes:
NO UI changes, can refer this tests file: src/core/server/ui_settings/settings/navigation.test.ts .

## Changelog

- feat: Add nav group enabled in chrome service

### Check List

- [x] All tests pass
  - [x] `yarn test:jest`
  - [x] `yarn test:jest_integration`
- [ ] New functionality includes testing.
- [ ] New functionality has been documented.
- [x] Update [CHANGELOG.md](./../CHANGELOG.md)
- [x] Commits are signed per the DCO using --signoff

### CODE REVIEW

File: src/core/public/chrome/chrome_service.mock.ts [hunk 1]
```diff
@@ -36,6 +36,7 @@ import { getLogosMock } from '../../common/mocks';
 const createSetupContractMock = () => {
   return {
     registerCollapsibleNavHeader: jest.fn(),
+    getUseCaseEnabled: jest.fn(),
```
Comment (SuZhou-Joe) [inline src/core/public/chrome/chrome_service.mock.ts:39]: Perhaps we need to call it `getNavGroupEnabled`? navGroup seems to be a term more acceptable by others from PR #7060

File: src/core/public/chrome/chrome_service.tsx [hunk 2]
```diff
@@ -495,4 +527,7 @@ export interface InternalChromeStart extends ChromeStart {
    * @internal
    */
   getHeaderComponent(): JSX.Element;
+  /**
+   * Get a boolean value to indicates whether use case is enabled
+   */
```
Comment (SuZhou-Joe) [inline src/core/public/chrome/chrome_service.tsx:532]: Seems no property is defined here.

File: src/core/public/chrome/chrome_service.mock.ts (hunk 1 above)
Comment (wanglam) [reply src/core/public/chrome/chrome_service.mock.ts:39]: Sure, will refactor all use case term to group in the next commit.

**Review by SuZhou-Joe — APPROVED** [review]

File: src/core/server/ui_settings/settings/navigation.ts [hunk 3]
```diff
@@ -80,5 +80,17 @@ export const getNavigationSettings = (): Record<string, UiSettingsParams> => {
...
+        defaultMessage: 'Enable nav group',
+      }),
+      value: false,
+      description: i18n.translate('core.ui_settings.params.navGroupEnabledDesc', {
+        defaultMessage: 'Used to control whether navigation items are grouped into use cases.',
+      }),
+      category: ['appearance'],
+      requiresPageReload: true,
+      schema: schema.boolean(),
+    },
```
Comment (ruanyl) [inline src/core/server/ui_settings/settings/navigation.ts:94]: I think the `name` and `description` should be simple and more intuitive to end users, can we simply say "Use new navigation" with description "Turn on/off the new navigation"? @lauralexis any suggestions?

**Comment by SuZhou-Joe** [comment]
Has been merged to #7060

**Pull request closed without merging**